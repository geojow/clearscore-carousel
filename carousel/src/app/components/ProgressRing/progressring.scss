.progress-ring {
  position: absolute;
  height: 370px;
  width: 370px;
}

circle {
  transform: rotate(-90deg);
  transform-origin: 50% 50%;
  animation: 1.4s ease-in-out both circle-animation;
  stroke-linecap: round;
}

@keyframes circle-animation {
  0% {
    stroke-dashoffset: 1112.1237993707869;
  }
}
