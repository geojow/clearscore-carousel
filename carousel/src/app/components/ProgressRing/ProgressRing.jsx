import React from 'react';
import PropTypes from 'prop-types';
import bemHelper from '../../utils/bem';
import './progressring.scss'

const cn = bemHelper({ block: "progress-ring" })

const ProgressRing = ({ progress }) => {

  const radius = 185
  const stroke = 4
  const normalisedRadius = radius - stroke * 2;
  const circumference = normalisedRadius * 2 * Math.PI;

  const strokeDashoffset = circumference - progress / 100 * circumference;

  return (
    <svg className={cn('')}>
      <circle
        fill="transparent"
        strokeWidth={stroke}
        strokeDasharray={circumference}
        style={{ strokeDashoffset, '--test': "black" }}
        r={normalisedRadius}
        cx={radius}
        cy={radius}
      />
    </svg>)
}

ProgressRing.propTypes = {
  progress: PropTypes.number.isRequired
}

export default ProgressRing;
