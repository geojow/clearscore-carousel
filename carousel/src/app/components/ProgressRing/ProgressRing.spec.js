import React from "react";
import { shallow } from "enzyme";
import ProgressRing from "./ProgressRing";

describe("ProgressRing", () => {
  let wrapper;

  beforeEach(() => (wrapper = shallow(<ProgressRing progress={0} />)));

  it("should render correctly", () => expect(wrapper).toMatchSnapshot());

  it("should render a <svg />", () => {
    expect(wrapper.find("svg").length).toEqual(1);
  });
});
