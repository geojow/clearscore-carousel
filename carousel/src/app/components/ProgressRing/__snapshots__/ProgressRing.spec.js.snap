// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressRing should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressRing
    progress={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <circle
        cx={185}
        cy={185}
        fill="transparent"
        r={177}
        strokeDasharray={1112.1237993707869}
        strokeWidth={4}
        style={
          Object {
            "--test": "black",
            "strokeDashoffset": 1112.1237993707869,
          }
        }
      />,
      "className": "progress-ring",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "cx": 185,
        "cy": 185,
        "fill": "transparent",
        "r": 177,
        "strokeDasharray": 1112.1237993707869,
        "strokeWidth": 4,
        "style": Object {
          "--test": "black",
          "strokeDashoffset": 1112.1237993707869,
        },
      },
      "ref": null,
      "rendered": null,
      "type": "circle",
    },
    "type": "svg",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <circle
          cx={185}
          cy={185}
          fill="transparent"
          r={177}
          strokeDasharray={1112.1237993707869}
          strokeWidth={4}
          style={
            Object {
              "--test": "black",
              "strokeDashoffset": 1112.1237993707869,
            }
          }
        />,
        "className": "progress-ring",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "cx": 185,
          "cy": 185,
          "fill": "transparent",
          "r": 177,
          "strokeDasharray": 1112.1237993707869,
          "strokeWidth": 4,
          "style": Object {
            "--test": "black",
            "strokeDashoffset": 1112.1237993707869,
          },
        },
        "ref": null,
        "rendered": null,
        "type": "circle",
      },
      "type": "svg",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
